utils/docs/gfix-library.1.in
.TH gfix-library 1 "@DATE@" "gEDA Project" @VERSION@

.SH NAME
gfix-library \- Fix gEDA Symbols Library

.SH SYNOPSIS
.B gfix-library
.RI  [ INFORMATION ] | [ OPTIONS ] [ DIRECTIVES ] Library

.SH DESCRIPTION
.B gfix-library
A script to assist in maintaining symbol libraries

This a wrapper script for gsymfix to repair a collection of symbol. gsymfix creates a new file with the same name and adds '.fix' as a suffix. By default gfix-library removes the orginal file and renames the new ".fix" file to the same as the orginal so as to replace the orginal symbol files. The --keep parameter can be used to retain the old symbol files if desired.

.SH QUICK
.B usage: [-h] | [-u] | [--version] | [-v] [-k] [-f <pattern>]

ex. 1) gfix-library incomming

ex. 2) gfix-library -k -f ic-?.sym incomming

.SH INFORMATION
Use of the following options displays information related to gfix-library
but does not inspect any symbols:
.IP "-h, --help"
Show information about the advanced usage of this command.
.IP "-u, --usage"
Displays examples on usage"
.IP "--version"
Displays software version information"

.SH OPTIONS
.IP "-v, --verbose"
Include additional information during processing.
.IP "-k, --keep"
keep orginal file, do not delete

.SH DIRECTIVES
.IP "-f, --filter <pattern>"
symbol file filter pattern is changed from '*.sym' to <pattern>

.SH HOMEPAGE
http://www.geda-project.org

.SH BUGS
You can report bugs at the Launchpad site of gfix-library:
https://bugs.launchpad.net/geda/gfix-library/+filebug

.SH AUTHOR
Wiley Edward Hill < wileyhill dot gmail dot com >

.SH SEE ALSO
.BR gsymfix (1)

.SH COPYRIGHT
.nf
Copyright \(co 2012-@YEAR@ Wiley Edward Hill.  License GPLv2+: GNU GPL
version 2 or later.  Please see the `COPYING' file included with this
program for full details.
.PP
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

.


NEWS
==============================================
 'gschem and Friends' Electronic Design Suite
==============================================

Copyright (C) 1998-2017 gEDA Developers

This file documents important user-visible changes in gEDA/gaf.  For
more information, please consult the `ChangeLog' file.

Notable changes in gEDA/gaf gEDA 2.1.1.1 - 04/01/19 Rampage Edition

* Expanded support for internationalization (i18n).

* All utilities now support internationalization (i18n).

* The visibility of toolbar tooltips can now be control using menu options.

* The context menu available when creating Polygon objects now includes
  an UNDO option allowing users to undo the previous section and resume
  path mode from the previous node.

* Scripting to support printing schematic documents from the command-line
  using gschem has been enhanced to support postscript or pdf output.
  Previously the default was changed from postscript to pdf. Postscript
  outputs could still be produced using menu options available in gschem.
  The output type is now automatically detected when using the command
  line interface by the file extension of the output file name. The
  default remains pdf if the output name is not given.

* Configuration function output-type has been renamed output-type.

* Libgedacairo now uses round end-caps when drawing buses, resulting in
  smoother transitions and improved appearance.

* A new keyword option hide-columns has been added to gattrib to allow
  user to designate a list attributes to be hidden in the GUI be default.
  A new Column Visibility dialog has been added to support user management
  of visible columns.

* A new path backend has been added to the gaf tool to allow scripting
  lanauages other than scheme to retrieve path information from libgeda.

* Improved appearence of various dialogs in gschem.

* gschem now logs exiting of the schematic editor if log-time is enabled.

* gschem now reports failure to attach attributes that aleady attached.

* gschem text properties dialog now supports operations to manipulate
  text case.

* gschem translate dialog now allows for translation of only selected
  objects. Additionally, the translate offset entry now accepts X and
  Y coordinates.

* Auxiliary buffers can now be explicitly cleared using a right context
  menu on the corresponding paste menu item.

* The suite now include template documents install to the user's template
  folder so the user can create new schematic and symbol documents from
  the system's context menu.

* Issues resolved:

* Libgedacairo has been enhanced to resolve issues with the end points of
  arcs shifting at different zoom levels. The shifting sometimes resulted
  in misalignment to adjoining lines at higher magnification levels.

* Extensive revisions were made to the Rampage Edition to build the suite
  on MinGW platforms. The Rampage Edition is now fully functional under
  native Win32 host systems including i18n support.

* The Rename Session dialog now fills in the previous session name,
  previously to dialog presented the user with an empty name field.

* The locked bit is not preserved when copying components in gschem.

* gschem now restarts text search when users change text in the Find Text
  dialog and click "find" or hits ENTER instead of closing the dialog.

* gschem now raises the main window when interacting with the Array Dialog.
  Previously, user could selected other application windows and return to
  the dialog and select the button that lowered the dialog but did not
  ensure the main window was in the foreground.

* An issue was resolved with controls not being initially enabled on the
  Attributes tab of the Preferences Dialog if the Attributes tab was the
  "saved" initial tab in gschem. Switching tabs or the view list controls
  re-enabled the disabled controls. The saved tab is now restored after
  settings are uploaded to the Preferences Dialog and this is reflected
  in the initial state of controls on the Attributes tab.

* An issue was resolved with resetting component-dialog-attributes when
  a second list was encountered in rc files. If a second attributes list
  was processed gschem would fault when the list was accessed or exiting
  gschem.

* The drc2 back end no longer reports duplicate slots for on components
  that do not have slots. This generally occurred when components have
  duplicate reference designators because these would typically show up
  using the same slot number.

* Libgeda no longer reports an oddity regarding symbol versions if the
  oddity was due to the symbol file being missing.

* The scheme file geda-deprecated-config.scm is no longer installed.

Notable changes in gEDA/gaf gEDA 2.1.1.1 - 11/12/17 Rampage Edition

* Expanded support for internationalization (i18n).

* The build system now support using the --with-included-gettext option to
  configure the suite to build and use an in-source international library.

* Font-sizes for widgets on GedaToolbars can now be specified in RC files.

* The toolbar layout in gschem has been improved to better support labels.

* The visibility of toolbar tooltips can now be control using RC keywords.

* The Slot Editor Dialog now incorporates a spinner control to adjust the
  slot number of slotted components. The range is restricted to valid slots
  for the selected component.

* New bitmaps have been added;

      * geda-deslect.png
      * geda-find-attribute
      * geda-properties.png
      * geda-trash-can.png

* libgedathon no longer over-writes existing gafrc when the declare_local_sym
  feature is used.

* libgedathon now supports internationalization (i18n).

Notable changes in gEDA/gaf gEDA 2.1.1.0 - 01/01/17 Rampage Edition

* Improvements were made to both the build system and source code to ensure
  gEDA-GAF builds more reliably on 64 bit platforms. Testing was performed
  using Ubuntu 12.04 and Ubuntu 14.04 64 bit host systems.


utils/scripts/gfix-library.sh
#!/bin/bash

# Author: Wiley Edward Hill
# Date: 10/20/12
#
# gEDA - GPL Electronic Design Automation
# gcheck-library - gEDA Symbol Library Batch Processor
# Copyright (C) 2012-2014 Wiley Edward Hill <wileyhill@gmail.com>
#
#------------------------------------------------------------------
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
#------------------------------------------------------------------

VER=1.0.0

KEEP=false
VERBOSE=false
QUIET=false

FILTER="*.sym"
LIBRARY="${@: -1}"
PROGRAM="gsymfix"
OPT=""
ERR_FILE_NOT_FOUND=2
ERR_BAD_ARGS=65

do_show_help()
{
   case $1 in
         "1") # Quick help
              echo Usage:   `basename $0` '[-h] | [-u] | [--version] | [-v] [-k] [-f <pattern>]' ;;
         "2") # Basic help
              echo
              echo "A script to assist in maintaining symbol libraries"
              echo Usage:   `basename $0` '[-options] [-d] [-f <pattern>] [library]'
              echo
              echo "Use the follow options to control behavior:"
              echo
              echo   '  -f | --filter <pattern>   filter pattern is changed to pattern'
              echo   ''
              echo   '  -h | --help      show this information'
              echo   '  -k | --keep      keep orginal file, do not delete'
              echo   '  -u | --usage     show basic usage format'
              echo   '  -v | --verbose   display extra information'
              echo   '       --version   Display version information'
              echo   ''
              echo ' Note: arguments are case sensitive. See documentation for more'
              echo ' detailed information.'
              echo ;;
          *)  echo "Help not avaliable for:$1" ;;
   esac
   exit 0
}

vecho()
{
   if [[ ! $1 = "" ]] ; then
     if $VERBOSE ; then
      echo $1
     fi
   fi
}

do_Assimilate_Arguments(){
   for Arg in $*; do
      case $Arg in
         --keep | -k) KEEP=true                                       ;;
         --help | -h) do_show_help "2"                                ;;
        --usage | -u) do_show_help "1"                                ;;
      --verbose | -v) VERBOSE=true  ; OPT="-vv"                       ;;
           --version) echo `basename $0` "Version $VER" ; exit        ;;
        --quiet | -q) QUIET=true                                      ;;
    	           *) case $LAST_ARG in
       --filter | -f) FILTER="$Arg"                                   ;;
                   *)                                                 ;;
                      esac
      esac
      LAST_ARG=$Arg
   done
}

do_fix_symbols (){
  exist=$(which $PROGRAM)
  if [[ ! $exist = "" ]] ; then
    for eachfile in $LIBRARY/$FILTER; do
      vecho "Checking $eachfile"
      $PROGRAM $OPT $eachfile
      if [ -f $eachfile.fix ] && [ !$KEEP ]; then
         rm -f $eachfile
         mv $eachfile.fix $eachfile
      fi
    done
  else
    echo "Error, recheck installation. Could not locate [$PROGRAM]"
  fi
}
# ----------------- Check for Empty Command Line ------------------
if [ "$#" -eq 0 ] ; then do_show_help "1"; exit $ERR_BAD_ARGS ; fi

# ----------------------- Parse Command Line ----------------------
do_Assimilate_Arguments $*

if [ "$LIBRARY" != "" ] && [ -d $LIBRARY ] ; then
    do_fix_symbols
else
    echo "Error, recheck argument: Could not access [$LIBRARY]"
fi
exit 0

